docs/docs/development/new-behavior.md

```diff
@@ -0,0 +1,383 @@
+---
+title: New Behavior
+sidebar_label: New Behavior
+---
+
+## Overview
+
+This document outlines how to develop a behavior for ZMK and prepare the changes for a pull request.
+
+The general process for developing behaviors is:
+
+1. [Create the behavior](#creating-the-behavior)
+   1. [Create the devicetree binding (`.yaml`)](#creating-the-devicetree-binding-yaml)
+   1. [Create the driver (`.c`)](#creating-the-driver-c)
+   1. [Update `app/CmakeLists.txt to include the new driver`](#updating-appcmakeliststxt-to-include-the-new-driver)
+   1. [Define common use-cases for the behavior (`.dtsi`) (Optional)](#defining-common-use-cases-for-the-behavior-dtsi-optional)
+1. [Test changes locally](#testing-changes-locally)
+1. [Document behavior functionality](#documenting-behavior-functionality)
+1. [Create a Pull Request for review and inclusion into the ZMK sources](#submitting-a-pull-request)
+
+:::info
+Before developing new behaviors, developers should have a working knowledge of the Embedded Linux Devicetree.
+The following resources are provided for those seeking further understanding:
+
+- [Embedded Linux Wiki - Device Tree Usage](https://elinux.org/Device_Tree_Usage)
+- [Zephyr Devicetree API](https://docs.zephyrproject.org/latest/build/dts/api/api.html)
+- [Zephyr Device Driver Model](https://docs.zephyrproject.org/latest/kernel/drivers/index.html)
+
+:::
+
+## Creating the Behavior
+
+### Creating the devicetree binding (`.yaml`)
+
+The properties of the behavior are listed in the behavior's devicetree binding, which comes in the form of a `.yaml` file. Devicetree bindings are stored in the directory `app/dts/bindings/behaviors/` and are labelled in lowercase, beginning with the prefix `zmk,behavior-`, and ending with the behavior's name, using dashes to separate multiple words. For example, the directory for the hold-tap's devicetree binding would be located at `app/dts/bindings/behaviors/zmk,behavior-hold-tap.yaml`.
+
+Using `zmk,behavior-hold-tap.yaml` as a reference:
+
+```
+# Copyright (c) 2020 The ZMK Contributors
```

Comment:
Should we keep the copyright headers here and in the `gresc` example? Unless it is needed for license purposes removing them would make this page more concise.

Reply:
Developers that are new to ZMK might not know to sign the files they create with the copyright headers for consistency across the codebase. Since Devicetree bindings, drivers, and pre-defined instances need to be signed with our `Copyright (c) XXXX The ZMK Contributors` headers, it would probably be useful to include some mention of the copyright stuff somewhere.

Reply:
That's true, it's probably worth pointing them out somewhere explicitly. Maybe we can add it to the last section (opening a pull request)? 

For these examples, we could say "[...] shown below as a reference (omitting copyright headers):"? But in any case, it's a minor issue and I don't mind the headers staying there either.

Reply:
I think the pull request section is a great place to put the copyright information. It would also help to keep things more concise, perhaps by implementing a tabbed menu that lets users copy and paste the correct copyright header depending on each file. Something similar to this section of the new shield guide:
![image](https://user-images.githubusercontent.com/27038994/165007225-cb68a59a-d823-4b94-87f4-44a20de6d7c1.png)

Reply:
Resolved in [aa17e24](https://github.com/zmkfirmware/zmk/commit/aa17e247bb23a6fefaf00be34aedf3ea376077ba)